inngest-core/src/main/kotlin/com/inngest/signingkey/SignatureVerification.kt

```diff
@@ -0,0 +1,104 @@
+package com.inngest.signingkey
+
+import com.inngest.InngestEnv
+import com.inngest.ServeConfig
+import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
+import java.time.Instant
+import javax.crypto.Mac
+import javax.crypto.spec.SecretKeySpec
+
+const val HMAC_SHA256 = "HmacSHA256"
+
+// Implementation of this is inspired by the pure Java example from https://www.baeldung.com/java-hmac#hmac-using-jdk-apis
```

Comment:
we could possibly replace this with a reputable library like BouncyCastle, but I chose the pure JDK way here